dascore/proc/inventory.py

```
@@ -0,0 +1,635 @@
+
+def _fill_from_intervals(distances, intervals, values, kind):
+    """
+    Return per-distance values of an interval track.
+
+    Coverage is half-open, matching the tracks themselves, with the
+    outermost covered endpoint included so the last channel of a path is
+    not silently uncovered. Point markers cover nothing.
+    """
+    fill = {"boolean": False, "numeric": np.nan}.get(kind, None)
+    out = np.full(len(distances), fill, dtype=object)
+    spans = [(lo, hi, value) for (lo, hi), value in zip(intervals, values) if lo < hi]
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)